### DISCUSSION TITLE
Language Switch Example Request

### DISCUSSION THREAD

**Discussion by taextreme** [SplashByte/animated_toggle_switch#23] [Q&A] [answered]
I currently try to implement language switch for my application. And I saw a gif example from pub.dev page. So, I want to ask how to use country flags for switch icons because there is no country flag icons build in or as a packages.

**Comment by maeddin** [comment, marked as answer]
I apologize for the late reply. I think I did it with the [country_icons](https://pub.dev/packages/country_icons) package. If you want an exact implementation, I can look it up too.

I didn't include the top two GIFs in my example because they require external packages.

EDIT: Here is the code for the flag which I used for the demo in the GIF. You can put this in the builder of the switch. Since I haven't touched this code for a long time, I can't guarantee that it works without problems, so feel free to give feedback if it works for you 😁.
If you want the code for the whole switch, I would have to assemble it myself first, since I have it all encapsulated.

```
Widget flag(Locale locale) {
  return FractionallySizedBox(
      widthFactor: 0.8,
      heightFactor: 0.8,
      child: Container(
          width: double.infinity,
          height: double.infinity,
          clipBehavior: Clip.antiAlias,
          decoration: BoxDecoration(shape: BoxShape.circle),
          child: Image.asset(
            'icons/flags/png/${countryCodeByLanguageCode(locale.languageCode)}.png',
            package: 'country_icons',
            fit: BoxFit.cover,
          )));
}
```
```
String? countryCodeByLanguageCode(String languageCode) {
  switch (languageCode) {
    case 'de':
      return 'de';
    case 'en':
      return 'gb';
  }
  return null;
}
```

In retrospect, I think you could also put the `Image` directly into the `BoxDecoration`, which might be more efficient.
Instead of the `FractionallySizedBox`, `Padding` will of course also work to create the border.

**Reply by taextreme** [reply]
Yes please. Right now i also try to use country_icons package. But the flag is show up as white rectangle on the switch.

**Reply by maeddin** [reply]
I will post my code tomorrow because it's too late for me now. Hopefully by then the rectangles will be enough for you. 😬

**Reply by taextreme** [reply]
That ok the switch work fine with only text. Country logo is just sprinkles on top that gave my project an edge.

**Reply by maeddin** [reply]
@taextreme I updated my answer :)

**Reply by maeddin** [reply]
@taextreme Is your question answered? Then can you please mark this comment as an answer?
